File: Samples for Parallel Programming with the .NET Framework/[C#][VB][C++][F#]-Samples for Parallel Programming with the .NET Framework/C# and C++ and F# and VB/C#,C++,F#,VB/Raytracer/Raytracer_VisualBasic/Scene.vb
'--------------------------------------------------------------------------
' 
'  Copyright (c) Microsoft Corporation.  All rights reserved. 
' 
'  File: Scene.vb
'
'--------------------------------------------------------------------------

Imports System.Collections.Generic

Public Class Scene
    Public Things As SceneObject()
    Public Lights As Light()
    Public Camera As Camera

    Public Sub New(ByVal things As SceneObject(), ByVal lights As Light(), ByVal camera As Camera)
        Me.Things = things
        Me.Lights = lights
        Me.Camera = camera
    End Sub

    Public Function Intersect(ByVal r As Ray) As IEnumerable(Of ISect)
        Dim objects = New List(Of ISect)()
        For Each obj In Things
            objects.Add(obj.Intersect(r))
        Next
        Return objects
    End Function
End Class

File: Samples for Parallel Programming with the .NET Framework/[C#][VB][C++][F#]-Samples for Parallel Programming with the .NET Framework/C# and C++ and F# and VB/C#,C++,F#,VB/Raytracer/Raytracer_CSharp/Scene.cs
//--------------------------------------------------------------------------
// 
//  Copyright (c) Microsoft Corporation.  All rights reserved. 
// 
//  File: Scene.cs
//
//--------------------------------------------------------------------------

using System.Collections.Generic;

namespace Microsoft.ParallelComputingPlatform.ParallelExtensions.Samples
{
    class Scene
    {
        public SceneObject[] Things;
        public Light[] Lights;
        public Camera Camera;

        public Scene(SceneObject[] things, Light[] lights, Camera camera) { Things = things; Lights = lights; Camera = camera; }

        public IEnumerable<ISect> Intersect(Ray r)
        {
            foreach (SceneObject obj in Things)
            {
                yield return obj.Intersect(r);
            }
        }
    }
}
